packages/docs/docs/advanced/canvas-operations.mdx

```diff
@@ -0,0 +1,110 @@
+
+You can see that for example the green trail turns yellow on the right side
+while remaining green on the left.
+
+## Masking and Composite Operations
+
+Composite Operations define how two layers are stacked ontop of each other. This
+way it is possible to define Masks. MDN has a
+[great visualisation of all Composite Operations](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/globalCompositeOperation#operations).
+
+If you are familiar with Adobe After Effects, you might know about
+[Track Mattes](https://helpx.adobe.com/after-effects/using/track-mattes-and-traveling-mattes.html).
+Track Mattes allow you to define a Background "Color" Layer, and a Foreground
+"Stencil" Layer. The Alpha Matte Layer for example will multiply the
+background's color by the stencil's alpha value.
+
+The same effect can be achieved in Motion Canvas with the help of Composite
+Operations using `source-in` and `source-out`.
+
+Create a new cached [`Node`](/api/2d/components/Node) that contains 2 children,
+the "stencil", and the "value" layers.
+
+```tsx
+const yOffset = createSignal(0);
+yield scene.add(<Background/>);
+yield scene.add(
+  <Node cache>
+    {/** This is your "stencil". It determines if something is visible or not */}
+    <AnimatedMotionCanvasIcon ... />
+    {/** This is your "value". It determines the output color if the stencil doesn't block it */}
+    <Img y={yOffset} src={exampleImage}
+      {/* 'source-in' for alpha stencil, 'source-out' for inverted alpha stencil*/}
+      compositeOperation={'source-in'}
+    />
+  </Node>
+);
+
+yield* yOffset(100, 2);
+yield* yOffset(-100, 2);
+```
+
+<AnimationPlayer name="canvas-operations-alpha-matte" />
+An example of how composite operations are used to create a mask. The Motion Canvas
+Icon acts as the stencil, the image in the back as the value.
+
+## Cached Nodes
+
+Both Filters and Composite Operations require a cached
+[`Node`](/api/2d/components/Node). Filters can set it automatically, while
+Composite Operations require you to set it explicitly on the parent
+[`Node`](/api/2d/components/Node).
+
+A cached [`Node`](/api/2d/components/Node) and its children are rendered on an
+offscreen canvas first, before getting added to the main scene.  
+For filters this is needed because they are applied to the entire canvas. By
+creating a new canvas and moving the elements that should get affected by the
+filters over, applying filters to the entire "new" canvas, and then moving back
+the result, you effectively only apply the filters to the moved elements.
+
+<!-- Prettier merges header with text -->
+<!-- prettier-ignore -->
+:::caution Performance Implications 
+If you care about playback performance, for
+example because you want to embed the Scene on a website, try to minimize the
+use of cached Nodes as each cached [`Node`](/api/2d/components/Node) means a separate offscreen canvas.
+
+<!-- prettier-ignore -->
+:::
```

Comment:
Admonitions should be formatted with new lines inbetween:
```suggestion
:::caution Performance Implications

If you care about playback performance, for example because you want to embed
the Scene on a website, try to minimize the use of cached Nodes as each cached
[`Node`](/api/2d/components/Node) means a separate offscreen canvas.

:::
```

Reply:
By the way, do we have any tests for this? 
How do additional canvases slow the playback?

Reply:
My main thought was that excessive use of filters would cause slowdown — having experienced this when working on some animations (this has been a week after the public release, so caching might have been different back then).

I tried to replicate but failed. I actually got the opposite result!

I tried this scene:
```tsx
const cacheNodes = true

export default makeScene2D(function* (scene) {
  scene.fill("#aaa")
  yield scene.add(
    <Node cache={cacheNodes}>
      {
        ([...new Array(1000).keys()]).map( (i) => {
          return <Icon filters={[blur(10)]} icon={"material-symbols:360"} cache={cacheNodes} x={useRandom().nextInt(-500, 500)} y={useRandom().nextInt(-500, 500)} fill={useRandom().nextInt(0, 0xFFFFFF)} size={40} />
        })
      } 
    </Node>
  )
  yield* waitFor(5);
});
```
This will hang for 2 or so seconds at the start of the scene, then render everything at 30 (target) FPS.
When I tried to set `cacheNodes` to `false`, I got 2 FPS. 

So I guess I will just remove the admonition.